src/keyboard/KeymapBindingChildren.tsx

```diff
@@ -0,0 +1,71 @@
+import { HidUsageLabel } from "./HidUsageLabel";
+
+export interface KeyBinding {
+  param1: number;
+  param2: number;
+}
+
+const renderIndicator = (numbers: 0|1|2|3 = 1): JSX.Element => (
+    <div className="absolute ml-0.5 h-full flex flex-col justify-evenly py-1">
+      {Array.from({ length: Math.max(1, numbers) }, (_, i) => (
+        <div 
+          key={i}
+          className="bg-gray-900 opacity-80 rounded-full" 
+          style={{
+            width: '.2rem',
+            height: numbers === 0 ? '0.8rem' : '.2rem',
+          }}
+        />
+      ))}
+    </div>
+  );
+
+export const getBindingChildren = (
+  behavior: string,
+  binding: KeyBinding,
+): JSX.Element | JSX.Element[] => {
+  if (behavior === "Layer-Tap") {
+    return [
+        <div className="relative text-sm">
+          {renderIndicator(1)}
+          <HidUsageLabel hid_usage={binding.param2}/>
+        </div>,
+        <div className="text-xs truncate relative">
+          {renderIndicator(0)}
+          <svg className="inline-block mb-0.5 mr-1" xmlns="http://www.w3.org/2000/svg" width="12" height="12" fill="currentColor" viewBox="0 0 16 16">
+            <path d="M8.235 1.559a.5.5 0 0 0-.47 0l-7.5 4a.5.5 0 0 0 0 .882L3.188 8 .264 9.559a.5.5 0 0 0 0 .882l7.5 4a.5.5 0 0 0 .47 0l7.5-4a.5.5 0 0 0 0-.882L12.813 8l2.922-1.559a.5.5 0 0 0 0-.882zm3.515 7.008L14.438 10 8 13.433 1.562 10 4.25 8.567l3.515 1.874a.5.5 0 0 0 .47 0zM8 9.433 1.562 6 8 2.567 14.438 6z"/>
+          </svg>
+          {binding.param1}
+        </div>
+    ];
+  }
+
+  if (behavior === "Mod-Tap") {
```

Comment:
We really can't do this. We need to be inspecting the metadata information to determine what to do here. Because:
* Users might have multiple mod taps with different names,
* Users might translate the display names to their language in the devicetree.
* General flexibility for custom behavior support in studio.

So... this is great to work on... *and* it *must* be metadata driven. For instance, we might choose this rendering style if both param1 and param2 are of parameter metadata type "HID usage".

Reply:
What do you think about the latest changes? Like that we don't depend on the name. It's not perfect but it's more generic.